### PR TITLE
docs(storage): updates the storage content for KRaft and node pools

### PR DESCRIPTION
**Pull request by PaulRMellor**
**Documentation**

Updates the storage content to split KRaft with node pools and ZooKeeper based config. 

Includes the following changes:

 * Updates to the introduction to storage and storage considerations
* Creates sections for storage config in KRaft (with node pool examples) and ZooKeeper specific storage config
* Moves node pool config related to storage to the configuring storage section
* Removes separate files on storage types to consolidate into single file
* Removes procedures to add and remove JBOD volumes and incorporates this content into the content related to JBOD storage
* Removes procedure to resize persistent volumes and  incorporates this content into the content related to persistent storage
* Updates to PVC naming conventions to reflect node pools

### Checklist

_Please go through this checklist and make sure all applicable tasks have been done_

- [ ] Write tests
- [ ] Make sure all tests pass
- [x] Update documentation
- [ ] Check RBAC rights for Kubernetes / OpenShift roles
- [ ] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [ ] Reference relevant issue(s) and close them after merging
- [ ] Update CHANGELOG.md
- [ ] Supply screenshots for visual changes, such as Grafana dashboards

### CODE REVIEW

File: documentation/assemblies/configuring/assembly-storage.adoc
[the remark this answers is not in the document]
Comment (PaulRMellor) [reply documentation/assemblies/configuring/assembly-storage.adoc:1]: Thanks @scholzj . I'll do some adjustments to make this clearer.

File: documentation/assemblies/configuring/assembly-storage.adoc [hunk 1]
```diff
@@ -6,24 +6,30 @@
...
-For example, tiered storage might include the following:
-+
-* Higher performance and higher cost block storage
-* Lower performance and lower cost object storage
+JBOD (Just a Bunch of Disks) storage allows you to configure your Kafka cluster to use multiple disks or volumes as ephemeral or persistent storage. 
 
-NOTE: The storage type cannot be changed after a Kafka cluster is deployed, though you can create and migrate to node pools with new storage specifications.
+.JBOD storage (multiple volumes)
+When specifying JBOD storage, you must still decide between using ephemeral or persistent volumes for each disk. 
+Even if you start with only one volume, using JBOD allows for future scaling by adding more volumes as needed.
```
Comment (scholzj) [inline documentation/assemblies/configuring/assembly-storage.adoc:22]: Could we add here something like `and that is why it is always recommended`?

File: documentation/assemblies/configuring/assembly-storage.adoc [hunk 2]
```diff
@@ -6,24 +6,30 @@
...
+
+NOTE: Persistent, ephemeral, and JBOD storage types cannot be changed after a Kafka cluster is deployed, though you can create and migrate to node pools with new storage specifications.
```
Comment (scholzj) [inline documentation/assemblies/configuring/assembly-storage.adoc:24]: Maybe we should clarify that you can add / remove volumes in the JBOD storage? Something like this (but feel free to rephrase it as needed):

```suggestion
NOTE: Persistent, ephemeral, and JBOD storage types cannot be changed after a Kafka cluster is deployed though you can add or remove volumes of different types from the JBOD storage.
You can also create and migrate to node pools with new storage specifications.
```

**Review by scholzj — COMMENTED** [review]
I left some smaller comments. But the structure looks good here. Thanks.

**Review by scholzj — APPROVED** [review]
Thanks!

**Pull request merged**